src/zenml/cli/service.py

```diff
@@ -0,0 +1,148 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""CLI for manipulating ZenML local and global config file."""
+import os
+from json import JSONDecodeError
+from typing import Optional
+
+import click
+
+from zenml.cli import utils as cli_utils
+from zenml.cli.cli import cli
+from zenml.config.global_config import GlobalConfiguration
+from zenml.io.utils import get_global_config_directory
+from zenml.logger import get_logger
+from zenml.services import ServiceRegistry, ServiceState
+from zenml.zen_service.zen_service import ZenService, ZenServiceConfig
+
+logger = get_logger(__name__)
+SERVICE_CONFIG_FILENAME = "ZenService.json"
+GLOBAL_ZENML_SERVICE_CONFIG_FILEPATH = os.path.join(
+    get_global_config_directory(), SERVICE_CONFIG_FILENAME
+)
+
+
+@cli.group()
+def service() -> None:
+    """ZenMl server."""
+
+
+@service.command("up", help="Start a daemon service running the zenml service")
+@click.option("--port", type=int, default=8000, show_default=True)
+@click.option("--profile", type=str, default=None)
+def up_server(port: int, profile: Optional[str]) -> None:
+    """Provisions resources for the zen service."""
+    if profile is not None:
+        profile_configuration = GlobalConfiguration().get_profile(profile)
+        if profile_configuration is None:
+            raise ValueError(f"Could not find profile of name {profile}")
+        service_config = ZenServiceConfig(
+            port=port, store_profile_configuration=profile_configuration
+        )
+    else:
+        service_config = ZenServiceConfig(port=port)
+
+    print(f"{service_config=}")
+    try:
+        with open(GLOBAL_ZENML_SERVICE_CONFIG_FILEPATH, "r") as f:
+            zen_service = ServiceRegistry().load_service_from_json(f.read())
+    except (JSONDecodeError, FileNotFoundError, ModuleNotFoundError):
+        zen_service = ZenService(service_config)
+
+    cli_utils.declare(
+        f"Provisioning resources for service "
+        f"'{zen_service.SERVICE_TYPE.name}'."
+    )
+
+    zen_service.start(timeout=30)
+
+    if zen_service.endpoint:
+        if zen_service.endpoint.status.port != port:
+            cli_utils.error(
+                f"You specified port={port} but the service is "
+                f"running at '{zen_service.endpoint.status.uri}' "
+                f"This can happen in case the specified port is "
+                f"in use or if the service was already running "
+                f"on port{zen_service.endpoint.status.port}' "
+                f"In case you want to change to port={port} "
+                f"shut down the service with "
+                f"`zenml service down -f` and restart it with "
+                f"a free port of your choice."
```

Comment:
```suggestion
                f"You specified port={port} but the service is "
                f"running at '{zen_service.endpoint.status.uri}' "
                "This can happen in case the specified port is "
                f"in use or if the service was already running "
                f"on port{zen_service.endpoint.status.port}' "
                f"In case you want to change to port={port} "
                "shut down the service with "
                "`zenml service down -f` and restart it with "
                "a free port of your choice."
```

I thought our linter would complain about this.

Reply:
now that @bcdurak  has taught me about textwrapper.dedent, I can just use a multiline string instead!